x-pack/legacy/plugins/rollup/public/crud_app/store/actions/create_job.js

```diff
@@ -78,11 +80,17 @@ export const createJob = (jobConfig) => async (dispatch) => {
     }));
   }
 
+  const deserializedJob = deserializeJob(newJob.data);
+
   dispatch({
     type: CREATE_JOB_SUCCESS,
-    payload: { job: deserializeJob(newJob.data) },
+    payload: { job: deserializedJob }
   });
 
+  if (jobConfig.startJobAfterCreation) {
+    dispatch(startJobs([jobConfig.id]));
```

Comment:
ah, I don't think I would have thought to do this from the client 🙂 I was originally thinking this would be implemented as a flag that gets passed to the job creation API, so the job would get started by the server code instead. this approach does make sense though! 👍

Reply:
That's a totally valid approach I hadn't considered 😄 . Still getting used to all the "mini" servers living with the plugins.

I guess the way it's currenty working leverages the existing code a bit more seamlessly albeit at the cost of an additional network request (client -> server -> es). Will definitely keep this in mind next time!